little-vdom.js

```diff
@@ -144,14 +144,17 @@ const diffChildren = (parentDom, newChildren, oldVNode) => {
     });
 
   // remove old children if there are any
-  oldChildren.map((oldChild) => {
-    const node = (oldChild._patched && oldChild._patched.dom) || oldChild.dom;
-    if (node) {
-      node.remove();
-    }
-  });
+  oldChildren.map(removePatchedChildren)
 
   return oldVNode;
 };
 
+function removePatchedChildren(child) {
+  const { _children = [], _normalizedChildren=[], _patched } = child
+  // remove children
+  _children.concat(_patched).map(c => c && removePatchedChildren(c))
+  // remove dom from _normalizedChildren or itself
+  _normalizedChildren.concat(child).map(i => i && i.dom && i.dom.remove())
```

Comment:
@luwes 
this line remove _normalizedChildren first, so remove text node first.
![image](https://user-images.githubusercontent.com/1826685/225807629-f41d813b-f165-42b9-9d7e-84982fede105.png)
after remove dom in _normalizedChildren, then `child.dom.outerHTML = "<li></li>"` not `" <li>d</li>"`